core/connection.js

```diff
@@ -170,15 +170,6 @@ Blockly.Connection.prototype.connect_ = function(childConnection) {
       if (!orphanBlock.outputConnection) {
```

Comment:
This check/exception is a guard for the next few lines of code.  It shouldn't be necessary now.  In fact, I don't think you need the if (parentConnection.type == Blockly.INPUT_VALUE) branch at all.